ArcDPS Boon Table/Tracker.cpp

```diff
@@ -29,12 +39,76 @@ void Tracker::addPlayer(ag* src, ag* dst)
 	}
 	else
 	{
-		std::unique_lock<std::mutex> lock(players_mtx);
-		players.emplace_back(id, characterName, accountName, subgroup, profession);
-		lock.unlock();
+		addNewPlayer(id, subgroup, profession, characterName, accountName);
+	}
+}
 
-		bakeCombatData();
+void Tracker::addNewPlayer(uintptr_t id,
+	uint8_t subgroup,
+	prof profession,
+	std::string characterName,
+	std::string accountName)
+{
+	std::unique_lock<std::mutex> lock(players_mtx);
+	players.emplace_back(id, characterName, accountName, subgroup, profession);
+	lock.unlock();
+
+	bakeCombatData();
+
+	//self has always the id 2000, not sure why, but this is easier than adding a new parameter to everything
```

Comment:
That is great, will change something in the killproof.me plugin with that knowledge :)

Reply:
Tested it a lot, and it seems consistent throughout patches, loading times, accounts, chars and maps. I just have no idea why it starts at 2000 and counts up from there.